MovieQuiz/Services/QuestionFactory.swift

```diff
@@ -39,15 +42,56 @@ class QuestionFactory: QuestionFactoryProtocol {
                                                           correctAnswer: false),
                                              QuizQuestion(image: "Vivarium",
                                                           text: "Рейтинг этого фильма больше чем 6?",
-                                                          correctAnswer: false)]
+                                                          correctAnswer: false)] */
 
     func requestNextQuestion() {
-        let index = (0..<questions.count).randomElement() ?? 0
-        let question = questions[safe: index]
-        delegate.didReceiveNextQuestion(question: question)
+        DispatchQueue.global().async { [weak self] in
+            guard let self = self else { return }
+            let index = (0..<self.movies.count).randomElement() ?? 0
+
+            guard let movie = self.movies[safe: index] else { return }
+
+            var imageData = Data()
+
+           do {
+               imageData = try Data(contentsOf: movie.resizedImageURL)
+            } catch {
+                print("Failed to load image")
+            }
+
+            let rating = Float(movie.rating) ?? 0
+
+            let text = "Рейтинг этого фильма больше чем 7?"
+            let correctAnswer = rating > 7
+
+            let question = QuizQuestion(image: imageData,
+                                         text: text,
+                                         correctAnswer: correctAnswer)
+
+            DispatchQueue.main.async { [weak self] in
+                guard let self = self else { return }
+                self.delegate?.didReceiveNextQuestion(question: question)
+            }
+        }
+    }
+
+    func loadData() {
+        moviesLoader.loadMovies { result in
```

Comment:
Здесь тоже нужно указывать weak self, чтобы контекст захватывался верно
```suggestion
        moviesLoader.loadMovies { [weak self] result in
```